### PR TITLE
Additional tunable consistency support for objects PUT, HEAD

### PR DESCRIPTION
**Pull request by parkerduckworth**
### What's being changed:

- Fix objects PUT tunable consistency bug
  - only default ALL consistency level was being used
- Implement HEAD (exists) tunable consistency
  - fix replication coordinator read exists bug

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

**Review by redouan-rhazouani — APPROVED** [review]

**Review by redouan-rhazouani — CHANGES_REQUESTED** [review]

File: usecases/replica/finder.go [hunk 1]
```diff
@@ -169,7 +169,7 @@ func (f *Finder) Exists(ctx context.Context, l ConsistencyLevel, shard string, i
 		if len(xs) == 1 {
 			x = xs[0]
 		}
-		if err == nil && len(xs) != 1 {
+		if err == nil && len(xs) < 1 {
```
[the remark this answers is not in the document]
Comment (parkerduckworth) [reply usecases/replica/finder.go:172]: Yeah I was thinking about this as well. What should the correct behavior be when multiple are returned for the same ID? Is this even possible? I guess we can just have another error condition when `len(xs) > 1`

[the remark this answers is not in the document]
Comment (redouan-rhazouani) [reply usecases/replica/finder.go:172]: Actually the correct behaviour is the previous one `if err == nil && len(xs) != 1 `.

[the remark this answers is not in the document]
Comment (parkerduckworth) [reply usecases/replica/finder.go:172]: We would probably want different error messages though, when `len(xs) == 0` and `len(xs) > 1`. For example, the previous error message of `digest read request: empty result` does not apply to the latter

[the remark this answers is not in the document]
Comment (redouan-rhazouani) [reply usecases/replica/finder.go:172]: Or we could just change the error message to something like.
`Errorf("malformed digest read request: expect one result got %d", len(XS)"), `

[the remark this answers is not in the document]
Comment (parkerduckworth) [reply usecases/replica/finder.go:172]: True, I will go with that

**Review by redouan-rhazouani — APPROVED** [review]

**Pull request merged**